### PR TITLE
Find installation method for analytics

### LINKED ISSUE #3184: Find out through what channel the SDK was installed by the user
Was the SDK installed through nuget?
For unity was the SDK installed through NPM or imported manually?

This is the kind of info that the metrics are after.

### PR DESCRIPTION
**Pull request by papafe**
Fixes #3184

### CODE REVIEW

**Comment by papafe** [comment]
I've timed it before. When the project is opened it takes about 1.8-2s to find the value, the following times about 0.3s

File: Realm/Realm.UnityWeaver/UnityWeaver.cs [hunk 1]
```diff
@@ -215,12 +256,15 @@ private static bool WeaveAssemblyCore(string assemblyPath, IEnumerable<string> r
...
+                    var installMethod = _installMethodTask.Task.Result;
+
                     var analyticsConfig = new Config
                     {
                         TargetOSName = targetOSName,
                         TargetFrameworkVersion = Application.unityVersion,
                         TargetFramework = framework,
-                        AnalyticsCollection = analyticsEnabled ? AnalyticsCollection.Full : AnalyticsCollection.Disabled
+                        AnalyticsCollection = analyticsEnabled ? AnalyticsCollection.Full : AnalyticsCollection.Disabled,
+                        InstallationMethod = installMethod
```
Comment (LaPeste) [inline Realm/Realm.UnityWeaver/UnityWeaver.cs:267]:
```suggestion
                        InstallationMethod = _installMethodTask.Task.Result
```

File: Realm/Realm.UnityWeaver/UnityWeaver.cs [hunk 2]
```diff
@@ -100,6 +109,38 @@ public static void Initialize()
...
+
+                        var source = package.source;
+                        installMethod = source switch
+                        {
+                            PackageSource.LocalTarball => "Manual",
+                            PackageSource.Registry => "NPM",
+                            _ => installMethod,
+                        };
+
+                        break;
```
Comment (LaPeste) [inline Realm/Realm.UnityWeaver/UnityWeaver.cs:135]:
```suggestion                        
                        if (package.name == UnityPackageName)
                        {
                            var source = package.source;
                            installMethod = source switch
                            {
                                PackageSource.LocalTarball => "Manual",
                                PackageSource.Registry => "NPM",
                                _ => installMethod,
                            };

                            break;
                        }
```

**Review by LaPeste — CHANGES_REQUESTED** [review]
Just a few nits on style but otherwise it looks good.

File: Realm/Realm.UnityWeaver/UnityWeaver.cs [hunk 3]
```diff
@@ -100,6 +109,36 @@ public static void Initialize()
...
+                var installMethod = "Unknown";
+
+                if (_listRequest.Status == StatusCode.Success)
+                {
+                    foreach (var package in _listRequest.Result)
+                    {
+                        if (package.name == UnityPackageName)
+                        {
+                            var source = package.source;
+                            installMethod = source switch
```
Comment (LaPeste) [inline Realm/Realm.UnityWeaver/UnityWeaver.cs:125]:
```suggestion
                            installMethod = package.source switch
```

**Review by LaPeste — APPROVED** [review]

**Comment by papafe** [comment]
@LaPeste looks fine to me!

**Pull request merged**